sys/sched_rr/sched_rr.c

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (C) 2020 TUBA Freiberg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys
+ * @{
+ *
+ * @file
+ * @brief       Round Robin Scheduler implementation
+ *
+ * @author      Karl Fessel <karl.fessel@ovgu.de>
+ *
+ * @}
+ */
+
+#include "thread.h"
+#include "xtimer.h"
+#include "clist.h"
+#include "sched_rr.h"
+
+static xtimer_t rr_timer;
+
+void schedule_rr(void * d){
+    // this lets the current thread give way
+    // simple RoundRobin
+    (void)d;
+    //this will run in an other threads context (xtimer/ISR)
+    //it can be started from any thread
+    //it will setup a timer to pull itself up
+    //this will try to reorder the current priority
+    //(put the current thread at the end of the list)
+    thread_t * active_thread = thread_get_active();
+    if(active_thread != NULL){
```

Comment:
Makes me wonder: Can't we simply turn off the sched tick when we enter the idle state?
And turn it back on when we leave idle?

Reply:
it would be possible to do that (turn off on idle) or to check when switching priority if there is more than one thread in that priority run queue. `active_thread == NULL` is not a case for idle but for a thread ended and no new one is scheduled yet